operator/pkg/apis/istio/v1alpha1/validation/validation.go

```diff
@@ -98,17 +98,18 @@ func deprecatedSettingsMessage(values *valuesv1alpha1.Values) string {
 }
 
 // validateFeatures check whether the config sematically make sense. For example, feature X and feature Y can't be enabled together.
-func validateFeatures(values *valuesv1alpha1.Values, _ *v1alpha1.IstioOperatorSpec) util.Errors {
+func validateFeatures(values *valuesv1alpha1.Values, iopSpec *v1alpha1.IstioOperatorSpec) util.Errors {
 	// When automatic mutual TLS is enabled, we check control plane security must also be enabled.
 	g := values.GetGlobal()
 	if g == nil {
 		return nil
 	}
-	m := g.GetMtls()
-	if m == nil {
-		return nil
+	mc := iopSpec.MeshConfig
+	val := true
+	if mc != nil && mc["enableAutoMtls"] != nil {
+		val = mc["enableAutoMtls"].(bool)
 	}
-	if m.GetAuto().GetValue() && !g.GetControlPlaneSecurityEnabled().GetValue() {
+	if val && !g.GetControlPlaneSecurityEnabled().GetValue() {
 		return []error{fmt.Errorf("security: auto mtls is enabled, but control plane security is not enabled")}
```

Comment:
can we just remove this whole check? GetControlPlaneSecurityEnabled doesn't actually do anything

Reply:
you mean the ControlPlaneSecurityEnabled setting? then maybe we should add it to deprecated fields?

Reply:
it controls mixer only actually. So it does do something I guess but not much

Reply:
ah ok then I think we can remove this validation, it creates a lot of warning noise to user

Reply:
but I keep the function there so later we can add other feature validations